pkg/cloud/cloud.go

```diff
@@ -574,8 +571,8 @@ func (c *cloud) DetachDisk(ctx context.Context, volumeID, nodeID string) error {
 		volume, err := c.getVolume(ctx, request)
 		klog.Infof("Check Volume state before detaching volume: %+v err: %+v",
 			volume, err)
-		if err == nil && !reflect.DeepEqual(volume, oscV1.Volume{}) {
-			if volume.GetState() != "" && volume.GetState() == "available" {
+		if err == nil {
+			if !volume.HasState() && volume.GetState() == "available" {
```

Comment:
Shouldn't it be `if volume.HasState() && volume.GetState() == "available"` ?

As it is two imbricated `if` with no additional instruction, you can just do `if err == nil && volume...`

Reply:
Indeed ! I'll will make the changes. Thank you